File: GameArchive_Version_01/app/src/main/java/com/melihakkose/gamearchive_version_01/MainActivity.java
package com.melihakkose.gamearchive_version_01;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    ListView listView;

    //Oyun adi ve id sini tutacagimiz listler.
    ArrayList<String> gameArray;
    ArrayList<Integer> idArray;
    ArrayAdapter arrayAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //list veiw ve listlerin tanimi
        listView=findViewById(R.id.listView);
        gameArray=new ArrayList<String>();
        idArray=new ArrayList<Integer>();

        //ArrayAdapter
        arrayAdapter=new ArrayAdapter(this,android.R.layout.simple_list_item_1,gameArray);
        listView.setAdapter(arrayAdapter); //listview' de gozukmesi icin gerekli

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Intent intent=new Intent(MainActivity.this,MainActivity2.class);
                intent.putExtra("id",idArray.get(i));
                intent.putExtra("activityNumb","first");
                startActivity(intent);
            }
        });

        //asagida yazdigimiz veri alma fonksiyonu
        getData();
    }

    //Veri cekme fonksşyonu
    public void getData(){
        try {
            SQLiteDatabase database=this.openOrCreateDatabase("Games",MODE_PRIVATE,null);

            Cursor cursor=database.rawQuery("SELECT * FROM games",null); //verileri secmek icin cursor

            int gameIx=cursor.getColumnIndex("gameName");
            int idIx=cursor.getColumnIndex("id");

            while (cursor.moveToNext()){
                gameArray.add(cursor.getString(gameIx));
                idArray.add(cursor.getInt(idIx));

            }
            arrayAdapter.notifyDataSetChanged(); //listede degisiklik olunca gostermek icin gerekli
            cursor.close();
        }catch (Exception e){
            e.printStackTrace();
        }
    }

    //Menu override fonksiyonu
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {

        //INFLATE.  XML olarak olusturdugumuz menuyu programimiza tanitmamiz gerekli.

        MenuInflater menuInflater=getMenuInflater();
        menuInflater.inflate(R.menu.add_game,menu);

        return super.onCreateOptionsMenu(menu);
    }
    //Menu override secilen item icin gerekli fonk
    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        /*MENU' den islem secildigi zaman yapilacak islem. Yeni Activity baslatmak olacak.
        * Ayni zamanda bu menun her item icin bir id' si olmali onlari da almaliyiz.*/


        if(item.getItemId()==R.id.add_game_item){
            Intent intent=new Intent(MainActivity.this,MainActivity2.class);
            intent.putExtra("activityNumb","second");
            startActivity(intent);
        }

        return super.onOptionsItemSelected(item);
    }
}

File: GameBook_Kotlin_SQLite/app/src/main/java/com/melihakkose/gamebook_kotlin_sqlite/MainActivity.kt
package com.melihakkose.gamebook_kotlin_sqlite

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.AdapterView
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.Exception

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val gameNameList=ArrayList<String>()
        val gameIdList=ArrayList<Int>()

        val arrayAdapter=ArrayAdapter(this,android.R.layout.simple_list_item_1,gameNameList)
        listView.adapter=arrayAdapter


        //VERILERI CEKME
        try{
            val database=this.openOrCreateDatabase("Games",Context.MODE_PRIVATE,null)

            val cursor=database.rawQuery("SELECT * FROM game",null)
            val nameIx=cursor.getColumnIndex("gameName")
            val idIx=cursor.getColumnIndex("id")

            while(cursor.moveToNext()){
                gameNameList.add(cursor.getString(nameIx))
                gameIdList.add(cursor.getInt(idIx))

            }
            cursor.close()
            arrayAdapter.notifyDataSetChanged()

        }catch (e:Exception){
            e.printStackTrace()
        }

        //OLUSTURDUGMUZ LISTEYE TIKLANDIGINDA NE OLACAGINI KODLUYORUZ
        listView.onItemClickListener= AdapterView.OnItemClickListener { parent, view, position, id ->
            val intent=Intent(this,DetailActivity2::class.java)
            intent.putExtra("info","list")
            intent.putExtra("id",gameIdList[position])
            startActivity(intent)
        }
    }

    //OLUSTURULAN MENU BAGLANIYOR
    override fun onCreateOptionsMenu(menu: Menu?): Boolean {

        //Inflater XML dosyalarini baglamak icin kullanilan kod kismi
        val menuInflater=menuInflater
        menuInflater.inflate(R.menu.add_game,menu)

        return super.onCreateOptionsMenu(menu)
    }

    //MENUDEN SECILEN ITEM ILE NE YAPILACAGINA DAIR KISIM
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        //Tiklaninca ne olacak?
        if(item.itemId==R.id.add_game_data){
            val intent=Intent(this,DetailActivity2::class.java)
            intent.putExtra("info","menu")
            startActivity(intent)
        }

        return super.onOptionsItemSelected(item)
    }



        /*
        VERITABANI KODLARI
        try {

            //Veritabani olusturma
            val myDatabase = this.openOrCreateDatabase("Games", Context.MODE_PRIVATE,null)

            // Veritabani ustunde calistirilacak kod
            //games adli tablo yok ise olustur ve name,releaseDate adlı sutunlar olustur
            myDatabase.execSQL("CREATE TABLE IF NOT EXISTS games (id INTEGER PRIMARY KEY,name VARCHAR, releaseDate INT)")

            //Veri ekleme kodu
            //myDatabase.execSQL("INSERT INTO games (name,releaseDate) VALUES('Red Dead Redemption 2',2018)")
            //myDatabase.execSQL("INSERT INTO games (name,releaseDate) VALUES('Cyberpunk 2077',2020)")
            // myDatabase.execSQL("INSERT INTO games (name,releaseDate) VALUES('Assassins Creed: Valhalla',2020)")

            //UPDATE
            //myDatabase.execSQL("UPDATE games SET name='Baldur s Gate 3' WHERE name='Assassins Creed: Valhalla'")

            //DELETE
            //myDatabase.execSQL("DELETE FROM games WHERE name='Baldur s Gate 3'")

            //Veri cekme -Filtreleme
            //val cursor=myDatabase.rawQuery("SELECT * FROM games WHERE id=2",null)
            val cursor=myDatabase.rawQuery("SELECT * FROM games ",null)

            val nameIx=cursor.getColumnIndex("name")
            val releaseIx=cursor.getColumnIndex("releaseDate")
            val idIx=cursor.getColumnIndex("id")

            while (cursor.moveToNext()){
                println("ID: "+ cursor.getString(idIx))
                println("Name: "+ cursor.getString(nameIx))
                println("Release Date: "+ cursor.getString(releaseIx))
            }
            cursor.close()

        }catch (e: Exception){
            e.printStackTrace()
        }

         */



}